keras/trainers/data_adapters/array_data_adapter_test.py

```diff
@@ -1,36 +1,39 @@
+import jax
 import numpy as np
 import pandas
 import pytest
 import tensorflow as tf
+import torch
 from absl.testing import parameterized
 
 from keras import backend
 from keras import testing
+from keras.testing.test_utils import named_product
 from keras.trainers.data_adapters import array_data_adapter
 
 
 class TestArrayDataAdapter(testing.TestCase, parameterized.TestCase):
     def make_array(self, array_type, shape, dtype="float32"):
+        x = np.array([[i] * shape[1] for i in range(shape[0])], dtype=dtype)
         if array_type == "np":
-            return np.ones(shape, dtype=dtype)
+            return x
         elif array_type == "tf":
-            return tf.ones(shape, dtype=dtype)
-        elif array_type == "backend":
-            if backend.backend() == "jax":
-                import jax
-
-                return jax.numpy.ones(shape, dtype=dtype)
-            elif backend.backend() == "torch":
-                import torch
-
-                return torch.tensor(np.ones(shape, dtype=dtype))
-            else:
-                return tf.ones(shape, dtype=dtype)
+            return tf.constant(x)
+        elif array_type == "jax":
+            return jax.numpy.array(x)
+        elif array_type == "torch":
+            return torch.as_tensor(x)
         elif array_type == "pandas":
-            return pandas.DataFrame(np.ones(shape, dtype=dtype))
+            return pandas.DataFrame(x)
 
-    @parameterized.parameters([("np",), ("tf",), ("backend",), ("pandas",)])
-    def test_basic_flow(self, array_type):
+    @parameterized.named_parameters(
+        named_product(
+            array_type=["np", "tf", "jax", "torch", "pandas"],
+            iterator_type=["np", "tf", "jax", "torch"],
```

Comment:
With this PR, all iterator types are tested with all backends (to continue testing the way it was tested before).

To speed this up a little bit, we could only test the iterator for the current backend. The CI will run the test with all backends anyway. Additionally, the only use case is getting the iterator for the current backend.